File: AndroidTai/app/src/main/java/com/tai/androidtai/domain/bean/DashBoardResponseBean.java
package com.tai.androidtai.domain.bean;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.SerializedName;

import java.util.List;

public class DashBoardResponseBean implements Parcelable {

    @SerializedName("meals")
    private List<MealBean> mList;

    private DashBoardResponseBean(Parcel in) {
    }

    public DashBoardResponseBean() {
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<DashBoardResponseBean> CREATOR = new Creator<DashBoardResponseBean>() {
        @Override
        public DashBoardResponseBean createFromParcel(Parcel in) {
            return new DashBoardResponseBean(in);
        }

        @Override
        public DashBoardResponseBean[] newArray(int size) {
            return new DashBoardResponseBean[size];
        }
    };

    public List<MealBean> getList() {
        return mList;
    }
}

File: AndroidTai/app/src/main/java/com/tai/androidtai/domain/bean/DashBoardResponseBean.kt
package com.tai.androidtai.domain.bean

import android.annotation.SuppressLint
import android.os.Parcel
import android.os.Parcelable

import com.google.gson.annotations.SerializedName

class DashBoardResponseBean private constructor(parcel: Parcel) : Parcelable {

    @SerializedName("results")
    var mResultBean: List<ResultBean>

    init {
        mResultBean = parcel.createTypedArrayList(ResultBean.CREATOR)
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeTypedList<ResultBean>(mResultBean)
    }

    override fun describeContents(): Int {
        return 0
    }

    fun getResultList(): List<ResultBean> {
        return mResultBean
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<DashBoardResponseBean> = object : Parcelable.Creator<DashBoardResponseBean> {
            override fun createFromParcel(parcel: Parcel): DashBoardResponseBean {
                return DashBoardResponseBean(parcel)
            }

            override fun newArray(size: Int): Array<DashBoardResponseBean?> {
                return arrayOfNulls(size)
            }
        }
    }
}

